<?xml version="1.0"?>
<launch>
  <rosparam command="load" file="$(find bitcraze_lps_estimator)/data/anchor_pos.yaml" />

  <node name="rviz" pkg="rviz" type="rviz"
        args="-d $(find bitcraze_lps_estimator)/data/rvizconfig.rviz"/>

  <node pkg="tf" type="static_transform_publisher" name="link1_broadcaster"
        args="1 0 0 0 0 0 1 world lps 100" />

  <group ns="crazyflie">
    <node pkg="crazyflie_driver" type="crazyflie_add" name="crazyflie_add" output="screen">
      <param name="uri" value="$(arg uri)" />
      <param name="tf_prefix" value="crazyflie" />
      <rosparam>
        genericLogTopics: ["log_ranges"]
        genericLogTopicFrequencies: [100]
        genericLogTopic_log_ranges_Variables: ["ranging.distance0", "ranging.distance1", "ranging.distance2", "ranging.distance3", "ranging.distance4", "ranging.distance5", "ranging.state"]
      </rosparam>
    </node>

    <node name="log_range" pkg="bitcraze_lps_estimator" type="log_range.py" />
    <node name="lps_lms" pkg="bitcraze_lps_estimator" type="lps_lms.py" />
    <node name="lps_viz" pkg="bitcraze_lps_estimator" type="lps_viz.py" />
  </group>

  <include file="$(find crazyflie_driver)/launch/crazyflie_server.launch" />
</launch>
